libfreerdp/core/gateway/http.c

```diff
@@ -362,7 +362,7 @@ static BOOL http_encode_print(wStream* s, const char* fmt, ...)
 {
 	char* str;
 	va_list ap;
-	size_t length, used;
+	int length, used;
 
```

Comment:
I don't understand what is done here, you change these variables to `int` and then they are cast to `size_t` in the rest of the function. What was wrong before ?

Reply:
@hardening The return value of `vsnprintf` is int, but other functions require `size_t` as argument.

Reply:
Then perhaps just casting the calls to `vsnprintf` would introduce less changes and would improve readability ?

Reply:
it is a return value of `vsnprintf`...